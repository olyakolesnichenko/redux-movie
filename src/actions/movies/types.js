export default Object.freeze({
    FETCH_MOVIES:              'FETCH_MOVIES',
    FETCH_MOVIES_SUCCESS:      'FETCH_MOVIES_SUCCESS',
    FETCH_MOVIES_FAIL:         'FETCH_MOVIES_FAIL',
    FETCH_FULL_MOVIE:          'FETCH_FULL_MOVIE',
    FETCH_FULL_MOVIE_SUCCESS:  'FETCH_FULL_MOVIE_SUCCESS',
    FETCH_FULL_MOVIE_FAIL:     'FETCH_FULL_MOVIE_FAIL',
    ADD_MOVIE:                 'ADD_MOVIE',
    ADD_MOVIE_SUCCESS:         'ADD_MOVIE_SUCCESS',
    ADD_MOVIE_FAIL:            'ADD_MOVIE_FAIL',
    DELETE_MOVIE:              'DELETE_MOVIE',
    DELETE_MOVIE_SUCCESS:      'DELETE_MOVIE_SUCCESS',
    DELETE_MOVIE_FAIL:         'DELETE_MOVIE_FAIL',
    FETCH_MY_LIST:             'FETCH_MY_LIST',
    FETCH_MY_LIST_SUCCESS:     'FETCH_MY_LIST_SUCCESS',
    FETCH_MY_LIST_FAIL:        'FETCH_MY_LIST_FAIL',
});
