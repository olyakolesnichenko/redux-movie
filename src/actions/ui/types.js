export default Object.freeze({
    INITIALIZE:            'INITIALIZE',
    START_FETCHING_MOVIES: 'START_FETCHING_MOVIES',
    STOP_FETCHING_MOVIES:  'STOP_FETCHING_MOVIES'
});
