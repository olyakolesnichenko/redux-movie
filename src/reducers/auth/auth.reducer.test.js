//core
import { Map } from 'immutable';

//Instruments
import reducer from './';
import types from 'actions/auth/types';

const initialState = Map({
    authenticated: false
});


describe('auth reducers:', () => {
    test(`'LOGIN_SUCCESS' `, () => {
        expect(
            reducer(initialState, {
                type:    types.LOGIN_SUCCESS
            })
        ).toMatchSnapshot();
    });
    test(`'SIGNUP_SUCCESS' `, () => {
        expect(
            reducer(initialState, {
                type:    types.SIGNUP_SUCCESS
            })
        ).toMatchSnapshot();
    });
    test(`'LOGOUT_SUCCESS' `, () => {
        expect(
            reducer(initialState, {
                type:    types.LOGOUT_SUCCESS
            })
        ).toMatchSnapshot();
    });
    test(`'default ' `, () => {
        expect(
            reducer(initialState, {
                type:    ''
            })
        ).toMatchSnapshot();
    });

});
