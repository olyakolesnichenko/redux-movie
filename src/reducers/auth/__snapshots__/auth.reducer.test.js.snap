// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducers: 'LOGIN_SUCCESS'  1`] = `
Immutable.Map {
  "authenticated": true,
}
`;

exports[`auth reducers: 'LOGOUT_SUCCESS'  1`] = `
Immutable.Map {
  "authenticated": false,
}
`;

exports[`auth reducers: 'SIGNUP_SUCCESS'  1`] = `
Immutable.Map {
  "authenticated": true,
}
`;

exports[`auth reducers: 'default '  1`] = `
Immutable.Map {
  "authenticated": false,
}
`;
